lib/QubitInformationObjectXmlCache.class.php

```diff
@@ -0,0 +1,217 @@
+   *
+   * @return null
+   */
+  protected function cacheXmlRepresentation($resource, $format)
+  {
+    $tempFile = tmpfile();
+    fwrite($tempFile, $this->generateXmlRepresentation($resource, $format));
+    $metadata = stream_get_meta_data($tempFile);
+    $this->storeXmlExport($metadata['uri'], $resource->id, $format);
+    fclose($tempFile);
+  }
+
+  /**
+   * Generate XML representation of information object.
+   *
+   * @param object  information object to cache
+   * @param string  format of XML ("dc" or "ead")
+   *
+   * @return string  XML representation
+   */
+  protected function generateXmlRepresentation($resource, $format)
+  {
+    exportBulkBaseTask::includeXmlExportClassesAndHelpers();
+    $rawXml = exportBulkBaseTask::captureResourceExportTemplateOutput($resource, $format);
+    return Qubit::tidyXml($rawXml);
+  }
+
+  /**
+   * Store XML representation of information object.
+   *
+   * @param string  path to temporary file containing XML
+   * @param integer  ID of information object to cache
+   * @param string  format of XML ("dc" or "ead")
+   *
+   * @return null
+   */
+  protected function storeXmlExport($filePath, $objectId, $format)
+  {
+    copy($filePath, $this->getFilePath($objectId, $format));
+    $skipLines = ($format == 'ead') ? 2 : 1;
+    $this->rewriteFileSkippingLines($filePath, $this->getFilePath($objectId, $format, true), $skipLines);
+  }
+
+  /**
+   * Get file path of an information object's XML representation.
+   *
+   * @param integer  ID of information object to cache
+   * @param string  format of XML ("dc" or "ead")
+   * @param boolean  where or not to store just the contents (no XML header lines)
+   *
+   * @return string  path to XML representation
+   */
+  public static function getFilePath($objectId, $format, $contentsOnly = false)
+  {
+    $filename = md5($objectId);
+    if ($contentsOnly)
+    {
+      $filename .= '_contents';
+    }
+    $filename .= '.'. strtolower($format) .'.xml';
+
+    $exportsPath = 'downloads' . DIRECTORY_SEPARATOR . 'exports';
+    return $exportsPath . DIRECTORY_SEPARATOR . strtolower($format) . DIRECTORY_SEPARATOR . $filename;
+  }
+
+  /**
+   * Get URL of an information object's XML representation.
+   *
+   * @param integer  ID of information object to cache
+   * @param string  format of XML ("dc" or "ead")
+   *
+   * @return string  URL of XML representation
+   */
+  public static function getPathForDownload($objectId, $format)
+  {
+    $path = self::getFilePath($objectId, $format);
+
+    if (file_exists(sfConfig::get('sf_web_dir') . DIRECTORY_SEPARATOR . $path))
+    {
+      return $path;
+    }
+
+    return null;
+  }
+
+  /**
+   * Rewrite file skipping lines.
+   *
+   * @param string  source file
+   * @param string  destination file
+   * @param integer  number of lines to skip
+   *
+   * @return string  URL of XML representation
+   */
+  protected function rewriteFileSkippingLines($source, $destination, $skipLines = 0)
```

Comment:
please elaborate on why we're skipping lines + rewriting the file, this function seems to be doin' some tricky stuff.